src/github.com/matrix-org/dendrite/roomserver/api/input.go

```diff
@@ -25,19 +27,14 @@ const (
 	// These events are state events used to authenticate other events.
 	// They can become part of the contiguous event graph via backfill.
 	KindOutlier = 1
-	// KindJoin event start a new contiguous event graph. The event must be a
-	// m.room.member event joining this server to the room. This must come with
-	// the state at the event. If the event is contiguous with the existing
-	// graph for the room then it is treated as a normal new event.
-	KindJoin = 2
```

Comment:
It looks like nobody is setting this correctly and I can't actually see how to use it correctly in the room-server.